src/main/webapp/js/index.js

```diff
@@ -12,81 +12,94 @@
+  }
+}
+
+/**
+ * Populate receipt display with newly queried receipts.
+ * @param {string} label User-entered label.
+ * @param {JSON Object} receipts Receipts returned from search query.
+ */
+function displayReceipts(label, receipts) {
   // If no receipts returned, display an error message. Else, display receipts.
-  if (Object.keys(receipts).length === 0) {
-    receiptsDisplayed.innerHTML =
-        '<div class="col-md-12 text-center error-message"><h3>' +
-        'Sorry, no results found for "' + label +
-        '". Please try your search again or try a different query.</h3></div>';
+  if (Object.keys(receipts).length == 0) {
+    createErrorMessageElement(label);
   } else {
     receipts.forEach((receipt) => {
-      receiptsDisplayed.innerHTML += createReceiptCardElement(receipt);
+      createReceiptCardElement(receipt);
     });
   }
 }
 
-
 /**
- * Returns an HTML string that represents a receipt card.
+ * Create receipt card based on existing HTML template.
  * This card displays transaction date, store name, trasaction total,
  * categories, receipt photo, and view/edit/delete buttons.
+ * @param {Receipt} receipt A Receipt object.
  */
 function createReceiptCardElement(receipt) {
   const categories = Array.from(receipt.categories);
-  const formattedHTML = '<div class="col-md-6">' +
-      '<div class="card mb-4 box-shadow">' +
-      '<p class="card-text align-self-end">' + receipt.timestamp + '</p>' +
-      '<p class="card-text">' + receipt.store + '</p>' +
-      '<p class="card-text">Total: $' + receipt.price + '</p>' +
-      '<div class="row">' +
-      '<div class="col-4 d-flex justify-content-center">' +
-      '<h4><span class="badge badge-pill badge-secondary">' + categories[0] +
-      '</span></h4>' +
-      '</div>' +
-      '<div class="col-4 d-flex justify-content-center">' +
-      '<h4><span class="badge badge-pill badge-secondary">' + categories[1] +
-      '</span></h4>' +
-      '</div>' +
-      '<div class="col-4 d-flex justify-content-center">' +
-      '<h4><span class="badge badge-pill badge-secondary">' + categories[2] +
-      '</span></h4>' +
-      '</div>' +
-      '</div>' +
-      '<img src=' + receipt.imageUrl +
-      ' alt="Receipt image" class="img-fluid receipt-img" />' +
-      '<div class="card-body">' +
-      '<div class="d-flex justify-content-between align-items-center">' +
-      '<div class="btn-group">' +
-      '<button type="button" class="btn btn-sm btn-outline-secondary">' +
-      'View</button' +
-      '><button type="button" class="btn btn-sm btn-outline-secondary">' +
-      'Edit</button' +
-      '><button type="button" class="btn btn-sm btn-outline-secondary">' +
-      'Delete</button>' +
-      '</div>' +
-      '</div>' +
-      '</div>' +
-      '</div>' +
-      '</div>';
-  return formattedHTML;
+
+  // Fill in template fields with correct information.
+  const templateScope = document.querySelector('#receipt-card-template');
```

Comment:
When is it better to use querySelector() vs getElementById()? I'm just curious since I'm not as familiar with javascript.

Reply:
I'm not sure that there really is a difference (stack overflow says querySelector is newer and the browser support might be different). I was following a tutorial to use templates and it used this method, so I just went with it